src/Parser/Mage/Fire/Modules/Features/Checklist.js

```diff
@@ -0,0 +1,178 @@
+    combustionSpellUsage: CombustionSpellUsage,
+    heatingUp: HeatingUp,
+    hotStreak: HotStreak,
+    mirrorImage: MirrorImage,
+    runeOfPower: RuneOfPower,
+
+    legendaryUpgradeChecker: LegendaryUpgradeChecker,
+    legendaryCountChecker: LegendaryCountChecker,
+    prePotion: PrePotion,
+    enchantChecker: EnchantChecker,
+  };
+
+  rules = [
+    new Rule({
+      name: 'Always be casting',
+      description: <Wrapper><em><b>Continuously chaining casts throughout an encounter is the single most important thing for achieving good DPS as a caster</b></em>. There shoule be no delay at all between your spell casts, it's better to start casting the wrong spell than to think for a few seconds and then cast the right spell. You should be able to handle a fight's mechanics with the minimum possible interruption to your casting. Some fights (like Argus) have unavoidable downtime due to phase transitions and the like, so in these cases 0% downtime will not be possible.</Wrapper>,
+      requirements: () => {
+        return [
+          new Requirement({
+            name: 'Downtime',
+            check: () => this.alwaysBeCasting.downtimeSuggestionThresholds,
+          }),
+          new Requirement({
+            name: 'Cancelled Casts',
+            check: () => this.cancelledCasts.cancelledCastSuggestionThresholds,
+          }),
+        ];
+      },
+    }),
+    new Rule({
+      name: 'Managing  your procs',
+      description: <Wrapper>Fire Mage is heavily dependent on correctly using your <SpellLink id={SPELLS.FIRE_BLAST.id}/> and <SpellLink id={SPELLS.PHOENIXS_FLAMES.id}/> guaranteed crit abilities to properly convert <SpellLink id={SPELLS.HEATING_UP.id}/> to <SpellLink id={SPELLS.HOT_STREAK.id}/>. Remember to only use your guaranteed crit spells to convert Heating Up to Hot Streak, and also remember to precede each instant cast <SpellLink id={SPELLS.PYROBLAST.id}/> with a hardcast of <SpellLink id={SPELLS.FIREBALL.id}/>, <SpellLink id={SPELLS.SCORCH.id}/>, or <SpellLink id={SPELLS.PYROBLAST.id}/> (If you have <ItemLink id={ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING.id}/>) to increase the chances of getting a new Heating Up or Hot Streak</Wrapper>,
+      requirements: () => {
+        return [
+          new Requirement({
+            name: "Fire Blast used without Heating Up",
+            check: () => this.heatingUp.fireBlastUtilSuggestionThresholds,
+          }),
+          new Requirement({
+            name: "Phoenix's Flames used without Heating Up",
+            check: () => this.heatingUp.phoenixFlamesUtilSuggestionThresholds,
+          }),
+          new Requirement({
+            name: "Hard Cast Before Hot Streak Util",
+            check: () => this.hotStreak.castBeforeHotStreakThresholds,
+          }),
+          new Requirement({
+            name: "Expired Hot Streaks",
+            check: () => this.hotStreak.expiredProcsThresholds,
+          }),
+          new Requirement({
+            name: "Wasted Crits",
```

Comment:
It might not be clear from this what a "wasted crit" is... I should really add something so requirements can have tooltips...

Reply:
i thought about trying to figure out how to add tooltips to it, but didnt end up doing it. Especially after the extremely long description i put on a couple of the rules. Tooltips would be better.